pym/portage/dbapi/vartree.py

```diff
@@ -3922,6 +3979,10 @@ def onerror(e):
 
 					relative_path = fpath[srcroot_len:]
 
+					# filter on INSTALL_MASK
+					if self._is_install_masked(relative_path, install_mask):
+						continue
```

Comment:
This filters the `_collision_protect` file list for the current package. However, it's asymmetric, because the following `_collision_protect` code is still going to report collisions if the current package installs a non-masked file that is install-masked by another package (isowner will return True even for install-masked files because they are still listed in CONTENTS):
```python
				for ver in mypkglist:
					if ver.isowner(f):
						isowned = True
						break```

Reply:
Is that bad? We don't allow two packages to own the same file. We don't have `protect-owned` to ignore collisions just because some developer is using INSTALL_MASK. As far as I'm concerned, we can remove this filter altogether from collision checks.

Reply:
It would be a change in behavior, which means a potential source of breakage for current users of INSTALL_MASK. In fact, at my workplace, it's very common to resolve file collisions via INSTALL_MASK. Dropping support for that could upset some people. If we don't plan to allow resolution of file collisions via INSTALL_MASK, we can advise users to implement an alternative to INSTALL_MASK via bashrc post_src_install hooks.

Reply:
Or teach them to use features specifically designed for this purpose such as `COLLISION_IGNORE`.

Reply:
Updated to cover this yet another awful obscure corner case of this idiotic package manager. Hope that's all and we can finally get this merged after 2 years of complete ignorance and then few days of torture on me.

Reply:
COLLISION_IGNORE doesn't resolve collisions, it ignores them. In many cases it's preferable to prevent collisions from occurring, since collisions can have negative consequences.

Reply:
Updated to cover this yet another awful obscure corner case of this idiotic package manager.

Collision protection is an extremely important feature, and any bugs that we introduce in collision protection could cause severe problems.

    Hope that's all and we can finally get this merged after 2 years of complete ignorance and then few days of torture on me.

I'm not trying to torture you, I just want to eliminate any bugs that I forsee.